test/e2e/rc.go

```diff
@@ -71,63 +95,46 @@ func TestBasicImage(c *client.Client, test string, image string) bool {
 			},
 		},
 	})
-	if err != nil {
-		glog.Infof("Failed to create replication controller %s: %v", name, err)
-		return false
-	}
+	Expect(err).NotTo(HaveOccurred())
 	// Cleanup the replication controller when we are done.
 	defer func() {
 		// Resize the replication controller to zero to get rid of pods.
 		controller.Spec.Replicas = 0
 		if _, err = c.ReplicationControllers(ns).Update(controller); err != nil {
-			glog.Errorf("Failed to resize replication controller %s to zero: %v", name, err)
+			Logf("Failed to resize replication controller %s to zero: %v", name, err)
 		}
 
 		// Delete the replication controller.
 		if err = c.ReplicationControllers(ns).Delete(name); err != nil {
-			glog.Errorf("Failed to delete replication controller %s: %v", name, err)
+			Logf("Failed to delete replication controller %s: %v", name, err)
 		}
-
-		// Report running time for test.
-		glog.Infof("Controller %s test took %v seconds", name, time.Since(testStart).Seconds())
 	}()
 
 	// List the pods, making sure we observe all the replicas.
 	listTimeout := time.Minute
 	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
 	pods, err := c.Pods(ns).List(label)
-	if err != nil {
-		glog.Errorf("Controller %s: Failed to list pods: %v", name, err)
-	}
+	Expect(err).NotTo(HaveOccurred())
 	t := time.Now()
 	for {
```

Comment:
If this becomes a common pattern (or already is, I haven't read through too much of the other test code yet), we should make a utility function to handle timeouts consistently. Something like util.Until(listTimeout, func() { })